### ISSUE TITLE
Copy button

### ISSUE THREAD

**Issue by ot-net** [sttatusx/vue-share-modal#1]
Copy button works on local but it doesn't copy the link when the app is on the server. I tried on Firefox and Chrome... The demo version in the read me works as well. What can be the issue? It is the exact same code...

How to remove the copy button so users use the link by selecting and copying instead of the button...? It can also solve the problem.

**Comment by freakingeek** [comment]
Hi,  
Could you please send me the error message?  
Is there any more information you could provide? 


@goktugerol-dev

**Comment by ot-net** [comment]
```
Property "modules" was accessed during render but is not defined on instance. 
  at <BaseTransition appear=false persisted=false mode=undefined  ... > 
  at <Transition name="scale" > 
  at <ShareModal show=true title="Proceso: Prueba de Proceso - Desarollo" footer-hint="Compartir con enlace: "  ... > 
  at <ProcessViewDescription button-hide= 
Object { display: "none", showComponent: true }
 publico=1 descripcion="<p>This is the title</p>"  ... > 
  at <KeepAlive> 
  at <ProcessView id="content" onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< 
Proxy { <target>: {…}, <handler>: {…} }
 > > 
  at <RouterView id="content" > 
  at <App>
```


This is the message I get when I click the button but it's probably because I removed the icons, not related to the button... I don't have any other message.
the link is dynamic...  If you select the link that appears and paste it in a new browser tab it works, the only thing is that the button doesn't keep the link in clipboard...

(I don't use the social media icons etc, I just need to have url so I removed the icons)

Here is how I made the link dynamically. (it had to be dynamic because shared link is another component)

template code:
(Notice the title, and link I made them dynamic)
```html
  <share-modal
    :show="showModal"
    :title="dynamicTitle"
    footer-hint="Compartir con enlace: "
    :link="dynamicLink"
    :variables="{
      fontFamily: 'Inter, sans-serif',
      red: '#0fa2f9',
      white: '#fefefe',
      primary: '#0fa2f9',
      primaryLight: '#ee6c4d08',
      secondary: '#3d5a80',
      secondaryLight: '#3d5a8096',
    }"
    mode="normal"
    @update:show="showModal = false"
  >
    <template v-for="(m, key) in modules" :key="key">
      <share-module tag="a" href="" :name="m.name">
        <component :is="m.component" />
      </share-module>
    </template>
```

Here is the script where I send the link as computed property:

```js
  computed: {
    dynamicLink() {
      return `${config.activeMod.FRONTEND_BASE_URL}/url/${this.procesoUuid}`;
    },
    dynamicTitle() {
      return `Proceso: ${this.title}`;
    },
```

I use a javascript file where I have my environment settings... Previously I was using it only to change the backend source, but I added a new one called frontend base url, for this.

Can it be because QA is **http** and not **https**?

So basically, it copies when I use on local, but doesn't copy when it is on the server... The link appears and it works if you copy manually..

**Comment by freakingeek** [comment]
Hello,
Thanks for providing this information. Maybe replacing the template with something like this can solve your problem. If not, let me know.

```vue
<share-modal
  :show="showModal"
  :title="dynamicTitle"
  footer-hint="Compartir con enlace: "
  :link="dynamicLink"
  :variables="{
    fontFamily: 'Inter, sans-serif',
    red: '#0fa2f9',
    white: '#fefefe',
    primary: '#0fa2f9',
    primaryLight: '#ee6c4d08',
    secondary: '#3d5a80',
    secondaryLight: '#3d5a8096',
  }"
  mode="normal"
  @update:show="showModal = false"
/>
```

@goktugerol-dev

**Comment by ot-net** [comment]
> 

Hi @sttatusx ! The issue persists.

I also tested in different ways..

1) I hard coded the first part of the dynamic link, didn't work
2) Just for test purpose, instead of using dynamic link I hard coded the link, as a string. And the issue persist, it doesn't copy.
3) I also tried with default setup, with icons and everything as how it is, it doesn't copy.

Is there any additional library that you miss to add in the documentation for the copy button? Like vue use / clipboard etc?

**Comment by freakingeek** [comment]
Nah, the copy button functionality is in pure JS. I just created a new vite project (Vue3) and installed the package, it did work...
Did you get any other errors in the console? Can you create a demo url?

@goktugerol-dev

**Comment by ot-net** [comment]
> 

I don't have any error, it's weird.

**Comment by freakingeek** [comment]
> > 
> 
> I don't have any error, it's weird.


I don't think your problem is related.

**Comment by ot-net** [comment]
Hi @sttatusx , just for feedback, today we built the app and sent to the production, and the button works after build.

**Comment by freakingeek** [comment]
> Hi @sttatusx , just for feedback, today we built the app and sent to the production, and the button works after build.

Thank you for your feedback 🙏
